src/jmc/compile/command/_flow_control.py

```diff
@@ -197,20 +198,41 @@ def __parse_switch_binary(min_: int, max_: int, count: str, datapack: DataPack,
 
 
 def parse_switch(scoreboard_player: ScoreboardPlayer,
-                 func_contents: list[list[str]], datapack: DataPack, name: str = SWITCH_CASE_NAME, start_at: int = 1) -> str:
+                 func_contents: list[list[str]], datapack: DataPack, name: str = SWITCH_CASE_NAME, start_at: int = 1, case_numbers: list[int | Literal["default"]] | None = None) -> str:
     """
     Create a binary tree for JMC switch-case
 
     :param scoreboard_player: Minecraft scoreboard objective to check the integer
     :param func_contents: List of function content(List of commands(string)) given by user
     :param datapack: Datapack object
     :param name: Private function's group name, defaults to SWITCH_CASE_NAME
+    :param case_numbers: List of case numbers provided (only matters with post-1.20.2 switch)
     :return: Minecraft function call to initiate switch case
     """
-    count = datapack.get_count(name)
-    __parse_switch_binary(start_at, len(func_contents) + start_at - 1, count,
-                          datapack, func_contents, scoreboard_player, name, start_at)
-    return f"function {datapack.namespace}:{DataPack.private_name}/{name}/{count}"
+    if case_numbers is None:
+        case_numbers = [*range(1, len(func_contents) + 1)]
+    func_count = datapack.get_count(name)
+    if datapack.version >= 16:
+        has_default = "default" in case_numbers
+        for (func, num) in zip(func_contents, case_numbers):
+            if has_default and num != "default":
+                func.append(f"scoreboard players set __found_case__ {datapack.var_name} 1")
+            datapack.add_raw_private_function(
+                name, func, f"{str(func_count)}/{num}")
+        datapack.add_raw_private_function(
+            name, [
+                f"$function {datapack.namespace}:{DataPack.private_name}/{name}/{func_count}/$(switch_key)"
+            ], f"{str(func_count)}/select")
+        return (
+            (f"scoreboard players set __found_case__ {datapack.var_name} 0\n" if has_default else "") +
+            f"execute store result storage {datapack.namespace}:{datapack.storage_name} switch_key int 1 run scoreboard players get {scoreboard_player.value[1]} {scoreboard_player.value[0]}" + # type: ignore
+            f"\nfunction {datapack.namespace}:{DataPack.private_name}/{name}/{func_count}/select with storage {datapack.namespace}:{datapack.storage_name}" +
+            (f"\nexecute unless score __found_case__ {datapack.var_name} matches 1 run function {datapack.namespace}:{DataPack.private_name}/{name}/{func_count}/default" if has_default else "")
+        )
+    else:
```

Comment:
Use a guard clause here instead of if else

Reply:
Done